### DISCUSSION TITLE
problem with parameters in url

### DISCUSSION THREAD

**Discussion by Jimmi08** [e107inc/e107#5170] [Q&A Core]
Hi,
company is using google ads. 
They are added parameter to website like ?gad_source= ***

After click on add URL is then siteurl?gad_source=999
and it displayed Page listing is turned off the page so it is going to page listing instead of displaying homeurl

Is something set wrong way?  
Facebook links work similar way....  

What to do with this?

**Comment by Jimmi08** [comment]
It is caused with this settings:

![image](https://github.com/e107inc/e107/assets/5429548/b5757151-9b37-4ae2-b02f-5c8388469890)


And I reported it in 2021 
https://github.com/e107inc/e107/issues/4565

Setting defaults (no page removal in configuration) is not helping because parameter in URL is changing layout and home layout is not detected correctly.

**Reply by tgtje** [reply]
maybe because of missing &gad as described [here](https://support.google.com/google-ads/answer/13327296?hl=en) ??

**Reply by Jimmi08** [reply]
it was the example, any URL with parameter will change the theme layout (try it on e107.nl) 
It is probably new URL tag for google,  
there are listed in e107 core those:  

`$trackers = array('fbclid','utm_source','utm_medium','utm_content','utm_campaign','elan', 'msclkid', 'gclid');`

But there are still other issues  with removing "page" from URL settings.

**Reply by tgtje** [reply]
@Jimmi08 normally i do only basics on NL. As Alex often is working on things, it might disrupt things.
As i also personally do not use g. ads i can not test things for comparison, and need of course server level entry.

Why i mention it ? (things i would test) :  a) mod rewerite (rewriteBase) has to be present.
b) what is the effect if i add an url in reference tab (page)(never used it so ?), and last c)  what happens iff the above settings On/Page are working with setting 3 Redirect to System not found page is ON (intended for diplaying true page(link) if not found.
That would sum up all things that pop in mind.

**Reply by Moc** [reply]
@Jimmi08 Cameron added the tags to the trackers array https://github.com/e107inc/e107/issues/5171. Is the issue solved now?

**Reply by Jimmi08** [reply]
No, it solved the issue but not this question.  This question is related to any parameters, not only tracker ones.

**Reply by Moc** [reply]
The question about url parameters is indeed still open (https://github.com/e107inc/e107/issues/4565), I don't have answer to that. 
Glad to hear the issue with the gad_source parameter is resolved.
